public/app/features/annotations/standardAnnotationSupport.ts

```diff
@@ -109,9 +109,17 @@ export const annotationEventNames: AnnotationFieldInfo[] = [
   {
     key: 'id',
   },
-  // { key: 'userId' },
-  // { key: 'login' },
-  // { key: 'email' },
+];
+
+// Properties we want passed though, but not exposed in the UI config builders
+const allAnnoEvents: AnnotationFieldInfo[] = [
```

Comment:
As far as I can see, it would make sense to add more items to `annotationEventNames` instead of creating `allAnnoEvents`. Also should probably add `prevState` and `data` to `AnnotationEvent` type instead of ignoring the type error with `as any`

Reply:
This is bit beyond me - @ryantxu know what needs to change perhaps?

(or if you know this is the right thing to do @domasx2  , feel free to push to this branch if you like)

Reply:
@domasx2 -- The issue here is that 'annotations' and 'alerts' use the same infrastructure, but annotations do not support these fields and we do not want to expand the generic annotations API with them yet.

TBH, I was surprised that alerting even uses the same code path! and this was the least intrusive way I could find to just let the values pass though.

I'll add more comments about this



not really.  The `standardAnnotaitonSupport` is made for annotations --